sstables/sstables.cc

```diff
@@ -2594,9 +2599,10 @@ future<checksum> sstable::read_checksum() {
 
         co_await crc_stream.close();
         maybe_rethrow_exception(std::move(ex));
+        _components->checksum = checksum->weak_from_this();
     });
 
-    co_return checksum;
+    co_return std::move(checksum);
```

Comment:
std::move() is unnecessary here (but not harmful). co_return of a local is an implicit move.